### ISSUE TITLE
9.0.0 to 9.0.1: Your data directory and your files are probably accessible from the Internet

### ISSUE THREAD

**Issue by crazy-max** [owncloud/core#23851]
Hi!

### Steps to reproduce
1. Migrate from ownCloud 9.0.0 to ownCloud 9.0.1 via apt-get update
2. Go to admin panel
3. Message displayed: "Your data directory and your files are probably accessible from the Internet. The .htaccess file is not working. We strongly suggest that you configure your web server in a way that the data directory is no longer accessible or you move the data directory outside the web server document root."
4. Check by trying access a file on my server: `http://myserver.com/owncloud/data/files/<username>/myfile.txt`

### Expected behaviour
Get a forbidden response

### Actual behaviour
All files in data are visible!

I have checked the `owncloud/data/.htaccess` and it's the same before update to 9.0.1 :

```
# Generated by ownCloud on 2016-04-07 19:19:12
# line below if for Apache 2.4
<ifModule mod_authz_core.c>
Require all denied
</ifModule>

# line below if for Apache 2.2
<ifModule !mod_authz_core.c>
deny from all
Satisfy All
</ifModule>

# section for Apache 2.2 and 2.4
IndexIgnore *
```

I have edited this file and commented the condition `<ifModule !mod_authz_core.c>` :

```
# Generated by ownCloud on 2016-04-07 19:19:12
# line below if for Apache 2.4
<ifModule mod_authz_core.c>
Require all denied
</ifModule>

# line below if for Apache 2.2
#<ifModule !mod_authz_core.c>
deny from all
Satisfy All
#</ifModule>

# section for Apache 2.2 and 2.4
IndexIgnore *
```

And i have a HTTP 403 Forbidden like expected. So i think it's a problem with `mod_authz_core`.
Here is my SSL vhost apache config :

```
<IfModule mod_ssl.c>
    <VirtualHost _default_:443>
        ServerName 10.0.0.1:443

        DocumentRoot /var/www/owncloud/
        <Directory /var/www/owncloud/>
            Options -Indexes +FollowSymLinks +MultiViews
            AllowOverride All
            Require all granted
        </Directory>

        ErrorLog ${APACHE_LOG_DIR}/error.log
        CustomLog ${APACHE_LOG_DIR}/access.log combined

        SSLEngine on
        SSLCertificateFile /etc/apache2/ssl/apache.crt
        SSLCertificateKeyFile /etc/apache2/ssl/apache.key
        <FilesMatch "\.(cgi|shtml|phtml|php)$">
            SSLOptions +StdEnvVars
        </FilesMatch>
        <Directory /usr/lib/cgi-bin>
            SSLOptions +StdEnvVars
        </Directory>

        <IfModule mod_headers.c>
            Header always set Strict-Transport-Security "max-age=15768000; includeSubDomains; preload"
        </IfModule>

        BrowserMatch "MSIE [2-6]" \
            nokeepalive ssl-unclean-shutdown \
            downgrade-1.0 force-response-1.0

        BrowserMatch "MSIE [17-9]" ssl-unclean-shutdown
    </VirtualHost>
</IfModule>
```

### Server configuration
**Operating system**:
Debian GNU/Linux 8 (jessie) 64bits

**Web server:**
Apache 2.4.10

**Database:**
MariaDB 10.1.12

**PHP version:**
PHP 5.6.17

**ownCloud version:** (see ownCloud admin page)
9.0.0.19

**Updated from an older ownCloud or fresh install:**
Upgrade from 9.0.0.19

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'updatechecker' => false,
  'instanceid' => 'octsmzfmuh8e',
  'passwordsalt' => '',
  'secret' => '',
  'trusted_domains' =>
  array (
    0 => '10.0.0.1',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'dbtype' => 'mysql',
  'version' => '9.0.1.3',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'owncloud',
  'dbpassword' => '',
  'installed' => true,
  'loglevel' => 0,
  'logdateformat' => 'Y-m-d H:i:s',
  'logtimezone' => 'Europe/Paris',
  'log_authfailip' => true,
  'cron_log' => true,
  'forcessl' => true,
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'enable_previews' => false,
  'trashbin_retention_obligation' => '7,15',
  'maintenance' => false,
);
```

**Are you using external storage, if yes which one:** local/smb/sftp/...

NO

**Are you using encryption:** yes/no

NO

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

NO

### Client configuration
**Browser:**
Firefox 45.0.1

**Operating system:**
Debian Jessie

**Comment by LukasReschke** [comment]
This seems more like a setup error. We only perform the check whether the data directory is accessible. Protecting it is up to the admin. (and in most cases it makes more sense to move it outside of the webroot anyways)

Are you sure that https://httpd.apache.org/docs/trunk/mod/mod_authz_core.html is installed? Also https://owncloud.org/support may be a better place to ask setup related questions.

Basically the following snippet is what should prevent the access:

```apache
<ifModule mod_authz_core.c>
Require all denied
</ifModule>
```

**Comment by crazy-max** [comment]
authz_core is enabled, this is strange :

```
$ a2enmod authz_core
Module authz_core already enabled
```

Do you have a best practice page to move data folder ?

Thanks

**Comment by crazy-max** [comment]
Ah okay gotcha : https://doc.owncloud.org/server/9.0/admin_manual/installation/installation_wizard.html#data-directory-location
